lib/swoosh/mailer.ex

```diff
@@ -56,19 +56,28 @@ defmodule Swoosh.Mailer do
 
       def __adapter__, do: @adapter
 
-      def deliver(%Swoosh.Email{from: nil}), do: raise ArgumentError, "expected \"from\" to be set"
-      def deliver(%Swoosh.Email{from: {_name, nil}}), do: raise ArgumentError, "expected \"from\" address to be set"
-      def deliver(%Swoosh.Email{to: nil}), do: raise ArgumentError, "expected \"to\" to be set"
-      def deliver(%Swoosh.Email{to: []}), do: raise ArgumentError, "expected \"to\" not to be empty"
-      def deliver(%Swoosh.Email{subject: nil}), do: raise ArgumentError, "expected \"subject\" to be set"
-      def deliver(%Swoosh.Email{html_body: nil, text_body: nil}) do
+      def deliver(email, config \\ [])
+      def deliver(%Swoosh.Email{from: nil}, _config), do:
+        raise ArgumentError, "expected \"from\" to be set"
+      def deliver(%Swoosh.Email{from: {_name, nil}}, _config), do:
+        raise ArgumentError, "expected \"from\" address to be set"
+      def deliver(%Swoosh.Email{to: nil}, _config), do:
+        raise ArgumentError, "expected \"to\" to be set"
+      def deliver(%Swoosh.Email{to: []}, _config), do:
```

Comment:
I find it pretty readable, but for now we can crack on.  I'm not happy about mixing the one line/multiline syntax though, `def .. do .. end`